airflow/www/views.py

```diff
@@ -803,6 +809,71 @@ def log(self):
             code=log, dag=dag, title=title, task_id=task_id,
             execution_date=execution_date, form=form)
 
+    def read_s3_log(self, log, remote_log_location):
+        remote_conn_id = conf.get('core', 'REMOTE_LOG_CONN_ID')
+        try:
+            from airflow.hooks import S3Hook
+            s3_hook = S3Hook(remote_conn_id)
+        except:
+            log += (
+                '*** Could not retrieve logs: '
+                'Could not create an S3Hook with connection id "{}". '
+                'Please make sure that airflow[s3] is installed and '
+                'the S3 connection exists.\n'.format(remote_conn_id))
+            return log
+
+        log += '*** Fetching log from S3: {}\n'.format(remote_log_location)
+        log += ('*** Note: remote logs are only available once '
+                'tasks have completed.\n')
+        s3_key = s3_hook.get_key(remote_log_location)
+        if s3_key:
+            log += '\n' + s3_key.get_contents_as_string().decode()
+        else:
+            log += '*** No log found on S3.\n'
+        return log
+
+    def read_gcs_log(self, log, remote_log_location):
+        remote_conn_id = conf.get('core', 'REMOTE_LOG_CONN_ID')
+        try:
```

Comment:
This is done in two places. Does it make sense to put it in one place as a util method?